### PR TITLE
Code style: Mark protected member vars as deprecated

### PR DESCRIPTION
**Pull request by elkuku**
Mark protected member vars as deprecated if they are prefixed with an underscore. Add proper pendant.
Package: application

I used:
`@deprecated use $XXX or declare as private`
Not sure as of which version to deprecate - please advise.
It should also be verified if the var could be declared as private, including the underscore.

This will fix 45 code style warnings.

### CODE REVIEW

**Comment by realityking** [comment]
Not sure if this is a good idea (and it's kinda cheating too). IMO we should deprecate them when we know whatever we wanna make them private or if we wanna rename them. Just adding the note we'll do this or that really isn't helpful for devs and just hides the problem.

**Comment by elkuku** [comment]
Well, yeah.. I believe it is not more than an "interim solution", I wouldn't call it "cheating" ;) but I believe it should be good enough to make "remember" that there is still some work to do.
When it comes to removing/changing the deprecated stuff some more careful thinking will be required on how to proceed. On the other hand, there are very few private methods/properties right now, so some of these might be good candidates.
It is also an attempt to continue @chdemko's work on #502

**Comment by eddieajau** [comment]
I know it's the PEAR standard, and it probably made sense when using PHP4, but I'm starting to wonder whether it makes sense to have the underscore rule for private variables.

**Comment by elkuku** [comment]
You mean private variables should also go without underscore ? I would totally agree as I personally hate those underscores.
It would be a slight modification of the sniff, and some more warnings.

The intent of this pull is to "mask" those warnings, so they won't appear in the "reports" @joomla-jenkins kindly generates for every pull request, while marking vars and methods as deprecated so they could be changed when there it's "time".

**Comment by eddieajau** [comment]
Yes, remove the underscores.  I came across a case where I was using some smart get methods, but it wouldn't work with private method name because they "had" to have an underscore prefix.  Probably a discussion for the list for another day.

**Comment by elkuku** [comment]
So would it be OK to continue with the "rest" ? Just to be absolutely clear...
1) Mark the **protected** vars and methods as deprecated, as in this pull.
2) Leave the **private** vars and methods as they are - for now ?

**Comment by eddieajau** [comment]
Yes.

File: libraries/joomla/application/component/helper.php [hunk 1]
```diff
@@ -24,6 +24,15 @@ class JComponentHelper
...
+	protected static $components = array();
+
+	/**
+	 * The component list cache
+	 *
+	 * @var    array
+	 * @since  11.1
+	 * @deprecated use $components declare as private
+	 */
 	protected static $_components = array();
```
Comment (realityking) [inline libraries/joomla/application/component/helper.php:36]: What's the use of this variable now? The code in this class doesn't use it.

Comment (eddieajau) [inline libraries/joomla/application/component/helper.php:36]: Probably isn't any use for it anymore.

**Pull request merged**